api/src/test/java/org/openmrs/api/context/DaemonTest.java

```diff
@@ -52,6 +58,52 @@ public void executeScheduledTask_shouldNotThrowErrorIfCalledFromATimerSchedulerT
 		Assert.assertTrue(new PrivateSchedulerTask(task).runTheTest());
 	}
 	
+	@Test 
+	public void createUser_shouldThrowWhenCalledOutsideContextDAO() throws Throwable {
+		try {
+			Daemon.createUser(new User(), "password", null);
+			Assert.fail("Should not be here, an exception should have been thrown in the line above");
+		}
+		catch (APIException e) {
+			Assert.assertTrue(e.getMessage().startsWith(Context.getMessageSourceService().getMessage("Context.DAO.only", new Object[] { this.getClass().getName() }, null)));
+		}
+	}
+	
+	@Test
+	public void createUser_shouldCreateUserWithRolesInContextDAO() throws Throwable {
+		// setup
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Jane", "X", "Doe"));
+		u.setUsername("jdoe");
+		u.getPerson().setGender("F");
+		
+		// replay
+		User createdUser = Context.getContextDAO().createUser(u, "P@ssw0rd", Arrays.asList("Provider", "Foobar Role"));
+		
+		// verif
+		Assert.assertNotNull(createdUser.getId());
+		Set<String> roleNames = createdUser.getRoles().stream().map(r -> r.getName()).collect(Collectors.toSet());
+		Assert.assertTrue(roleNames.contains("Provider"));
+		Assert.assertFalse(roleNames.contains("Foobar Role")); // a bogus role name has just no impact on the created user
+	}
+	
+	@Test 
```

Comment:
Any reason why you did not use the expected exception annotation?

Reply:
The other alternative is this: https://github.com/openmrs/openmrs-core/blob/917caf2265c70e0e32cd5f92644c9dcc47f6b845/api/src/test/java/org/openmrs/api/AdministrationServiceTest.java#L69

Reply:
@dkayiwa great, I actually had never used this. Coming in the next commit!